fe/fe-core/src/main/java/com/starrocks/scheduler/PartitionBasedMaterializedViewRefreshProcessor.java

```diff
@@ -364,49 +364,81 @@ private boolean needToRefreshNonPartitionTable() {
+            throws AnalysisException {
+        String start = properties.get(TaskRun.PARTITION_START);
+        String end = properties.get(TaskRun.PARTITION_END);
+        boolean force = Boolean.parseBoolean(properties.get(TaskRun.FORCE));
+        if (force && start == null && end == null) {
+            return Sets.newHashSet(materializedView.getPartitionNames());
+        }
         Set<String> needRefreshMvPartitionNames = Sets.newHashSet();
-
         PartitionInfo partitionInfo = materializedView.getPartitionInfo();
         if (partitionInfo instanceof SinglePartitionInfo) {
             // for non-partitioned materialized view
-            if (needToRefreshNonPartitionTable()) {
-                needRefreshMvPartitionNames.addAll(materializedView.getPartitionNames());
+            if (force || needToRefreshNonPartitionTable()) {
+                return Sets.newHashSet(materializedView.getPartitionNames());
             }
         } else if (partitionInfo instanceof ExpressionRangePartitionInfo) {
             Expr partitionExpr = getPartitionExpr();
             Pair<Table, Column> partitionTableAndColumn = getPartitionTableAndColumn(snapshotBaseTables);
             Table partitionTable = partitionTableAndColumn.first;
+            Set<String> mvRangePartitionNames = SyncPartitionUtils.getPartitionNamesByRange(materializedView, start, end);
 
             if (needToRefreshNonPartitionTable(partitionTable)) {
                 // if non partition table changed, should refresh all partitions of materialized view
-                needRefreshMvPartitionNames.addAll(materializedView.getPartitionNames());
-                return needRefreshMvPartitionNames;
+                return Sets.newHashSet(materializedView.getPartitionNames());
             }
-
             // check partition table
             if (partitionExpr instanceof SlotRef) {
-                Set<String> baseChangedPartitionNames = materializedView.getNeedRefreshPartitionNames(partitionTable);
-                for (String basePartitionName : baseChangedPartitionNames) {
-                    needRefreshMvPartitionNames.addAll(mvContext.baseToMvNameRef.get(basePartitionName));
-                }
+                return getMVPartitionNamesToRefreshByRangePartitionNamesAndForce(mvRangePartitionNames, force);
             } else if (partitionExpr instanceof FunctionCallExpr) {
-                // check if there is a load in the base table and add it to the refresh candidate
-                Set<String> baseChangedPartitionNames = materializedView.getNeedRefreshPartitionNames(partitionTable);
-                for (String baseChangedPartitionName : baseChangedPartitionNames) {
-                    needRefreshMvPartitionNames.addAll(mvContext.baseToMvNameRef.get(baseChangedPartitionName));
-                }
+                needRefreshMvPartitionNames = getMVPartitionNamesToRefreshByRangePartitionNamesAndForce(
```

Comment:
sorry. calcPotentialRefreshPartition has the logic of mapping. here just collecting the original mv partitions names is ok. The mapping logic here is not necessary.

Reply:
yes, there is no mapping logic, just collecting the original mv partitions names  by range and force.